imap/imapd.c

```diff
@@ -7875,6 +7875,25 @@ static void cmd_rename(char *tag, char *oldname, char *newname, char *location,
              imapd_userisadmin) {
 
         rename_user = 1;
+
+        // we can't rename users if the new inbox already exists!
+        mbentry_t *destmbentry = NULL;
+        r = mboxlist_lookup_allow_all(newmailboxname, &destmbentry, NULL);
+        if (r != IMAP_MAILBOX_NONEXISTENT) {
+            mboxlist_entry_free(&destmbentry);
+            r = IMAP_MAILBOX_EXISTS;
```

Comment:
This could mask other weird errors couldn't it? Maybe they'll all get log entries, but it's not clear.
Should we log something if r != IMAP_MAILBOX_EXISTS already?